server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateCleanupManager.java

```diff
@@ -221,6 +228,14 @@ void deleteClusterMetadata(
                             attribute -> addStaleGlobalMetadataPath(attribute.getUploadedFilename(), filesToKeep, staleGlobalMetadataPaths)
                         );
                 }
+                if (clusterMetadataManifest.getCodecVersion() >= ClusterMetadataManifest.CODEC_V3) {
+                    clusterMetadataManifest.getIndicesRouting().forEach(uploadedIndicesRouting -> {
+                        if (filesToKeep.contains(uploadedIndicesRouting.getUploadedFilename()) == false){
+                            staleIndexRoutingPaths.add(new BlobPath().buildAsString() + uploadedIndicesRouting.getUploadedFilename());
```

Comment:
Why do we need `new BlobPath().buildAsString()`?

Reply:
It is an empty path, removed it.